OpenRA.Mods.Common/Orders/PlaceBuildingOrderGenerator.cs

```diff
@@ -172,89 +231,58 @@ bool AcceptsPlug(CPos cell, PlugInfo plug)
 		protected override IEnumerable<IRenderable> Render(WorldRenderer wr, World world) { yield break; }
 		protected override IEnumerable<IRenderable> RenderAboveShroud(WorldRenderer wr, World world)
 		{
-			var topLeft = viewport.ViewToWorld(Viewport.LastMousePos + topLeftScreenOffset);
-			var centerPosition = world.Map.CenterOfCell(topLeft) + centerOffset;
-			var rules = world.Map.Rules;
-
-			foreach (var dec in actorInfo.TraitInfos<IPlaceBuildingDecorationInfo>())
-				foreach (var r in dec.Render(wr, world, actorInfo, centerPosition))
-					yield return r;
+			var topLeft = TopLeft;
+			var footprint = new Dictionary<CPos, PlaceBuildingCellType>();
+			var activeVariant = variants[variant];
+			var actorInfo = activeVariant.ActorInfo;
+			var buildingInfo = activeVariant.BuildingInfo;
+			var plugInfo = activeVariant.PlugInfo;
+			var lineBuildInfo = activeVariant.LineBuildInfo;
+			var preview = activeVariant.Preview;
 
```

Comment:
`Orders/PlaceBuildingOrderGenerator.cs(242,1): error SA1507: Code should not contain multiple blank lines in a row`